Ik 36

adrielFab opened the pull request:
This is the user story for view routes. This user story includes the new homepage, when the user first logs in. It also includes a navigation drawer which includes the account, the inbox, the settings, and the logout. A FAB is added at the bottom right for the addition of a route.

adrielFab committed:
IK-62 The basic layout of HomePge was created

The widgets used is a scrollview. Also, android.manifest was updated to load the page directly

adrielFab committed:
IK-62 The methods to retrieve the routes and display it to the UI are implemented in the HomePage.java

adrielFab committed:
IK-62 Navigation Drawer has been implemented

Many files have been modified since a menu was to be created and the configuration of the manifest file needs to specify which activities should include an action bar

adrielFab committed:
IK-62 Reimplemented the routes to be viewed after the implementation of the navigation drawer

adrielFab committed:
IK-62 Floating Action Bar was added in the home page to follow the UI design

adrielFab committed:
IK-62 Included documentation in the file Homepage.java

SaifMahabub commented on client/app/src/main/java/com/example/mrides/Activity/HomePage.java:
```diff
@@ -0,0 +1,149 @@
+     * @param savedInstanceState
+     */
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
```
Missing space after opening bracket.

handriamanantena commented on client/app/src/main/java/com/example/mrides/Activity/HomePage.java:
```diff
@@ -76,27 +83,27 @@ public void onClick(View view) {
-//            linearLayout.addView(route, params);
-//
-//        }
+        LinearLayout linearLayout = (LinearLayout) findViewById(R.id.layoutScrollUnmatch);
```
This segment of code is identical to the above section of code. Code duplication. You should create a helper private method for this segment of code.

handriamanantena replied:
Redundant code. Similar code above. Create a helper method for this block of code.

SaifMahabub commented on client/app/src/main/java/com/example/mrides/Activity/HomePage.java:
```diff
@@ -0,0 +1,149 @@
+        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+
+        navigationView = (NavigationView) findViewById(R.id.navvy);
+        navigationView.setNavigationItemSelectedListener(new NavigationView.OnNavigationItemSelectedListener() {
```
Missing space after opening bracket.

handriamanantena replied:
avoid using anonymous classes within parameters. It makes it difficult to reference variables within the Activities, because the references will have to be declared as final. To solve this problem the activity should implement NavigationView.OnNavigationItemSelectedListener as an interface and then override the method with the code you have within the anonymous class.

SaifMahabub commented on client/app/src/main/java/com/example/mrides/Activity/HomePage.java:
```diff
@@ -0,0 +1,149 @@
+        navigationView = (NavigationView) findViewById(R.id.navvy);
+        navigationView.setNavigationItemSelectedListener(new NavigationView.OnNavigationItemSelectedListener() {
+            @Override
+            public boolean onNavigationItemSelected(@NonNull MenuItem item) {
```
Missing space after opening bracket.

SaifMahabub commented on client/app/src/main/java/com/example/mrides/Activity/HomePage.java:
```diff
@@ -0,0 +1,149 @@
+        navigationView.setNavigationItemSelectedListener(new NavigationView.OnNavigationItemSelectedListener() {
+            @Override
+            public boolean onNavigationItemSelected(@NonNull MenuItem item) {
+                switch (item.getItemId()) {
```
Missing space after opening bracket.

SaifMahabub commented on client/app/src/main/java/com/example/mrides/Activity/HomePage.java:
```diff
@@ -0,0 +1,149 @@
+     */
+    public boolean onOptionsItemSelected(MenuItem item) {
+
+        if (toggle.onOptionsItemSelected(item)) {
```
Missing space after opening bracket.

SaifMahabub commented on client/app/src/main/java/com/example/mrides/Activity/HomePage.java:
```diff
@@ -0,0 +1,149 @@
+    /**
+     * Displays the matched routes of the user
+     */
+    public void createMatchedRoutes() {
```
Missing space after opening bracket.

SaifMahabub commented on client/app/src/main/java/com/example/mrides/Activity/HomePage.java:
```diff
@@ -0,0 +1,149 @@
+    public void createMatchedRoutes() {
+        LinearLayout linearLayout = (LinearLayout) findViewById(R.id.layoutScrollMatch);
+
+        for (int i = 0; i < matchedRoutes.length; i++) {
```
Missing space after opening bracket.

SaifMahabub commented on client/app/src/main/java/com/example/mrides/Activity/HomePage.java:
```diff
@@ -0,0 +1,149 @@
+    /**
+     * Displays the unmatched routes of the user
+     */
+    public void createUnmatchedRoutes() {
```
Missing space after opening bracket.

SaifMahabub commented on client/app/src/main/java/com/example/mrides/Activity/HomePage.java:
```diff
@@ -0,0 +1,149 @@
+    public void createUnmatchedRoutes() {
+        LinearLayout linearLayout = (LinearLayout) findViewById(R.id.layoutScrollUnmatch);
+
+        for (int i = 0; i < unmatchedRoutes.length; i++) {
```
Missing space after opening bracket.

SaifMahabub commented on client/app/src/main/java/com/example/mrides/Activity/HomePage.java:
```diff
@@ -0,0 +1,149 @@
+     * @param button
+     * @return params
+     */
+    public LinearLayout.LayoutParams styleButton(Button button){
```
Missing space after opening bracket.

handriamanantena commented on client/app/src/main/java/com/example/mrides/Activity/HomePage.java:
```diff
@@ -76,27 +83,27 @@ public void onClick(View view) {
-//            linearLayout.addView(route, params);
-//
-//        }
+        LinearLayout linearLayout = (LinearLayout) findViewById(R.id.layoutScrollMatch);
```
Redundant code. Similar code below. Create a helper method for this block of code.

adrielFab committed:
IK-62 Refactored the paranthesis to follow coding convention and refactored the Navigation View method

Navigation view is now being implemented in Homepage.java which overrides the method

adrielFab committed:
Merge branch 'dev' into IK-36
